librz/core/cmd/cmd_debug.c

```diff
@@ -3904,6 +3796,101 @@ RZ_IPI int rz_cmd_debug_continue_until(void *data, const char *input) {
 	return RZ_CMD_STATUS_OK;
 }
 
+// ds
+RZ_IPI RzCmdStatus rz_cmd_debug_step_handler(RzCore *core, int argc, const char **argv) {
+	const int times = argc > 1 ? (int)rz_num_math(core->num, argv[1]) : 1;
+	rz_core_debug_step_one(core, times);
+	rz_core_dbg_follow_seek_register(core);
+	return RZ_CMD_STATUS_OK;
+}
+
+// dsb
+RZ_IPI RzCmdStatus rz_cmd_debug_step_back_handler(RzCore *core, int argc, const char **argv) {
+	const int times = argc > 1 ? (int)rz_num_math(core->num, argv[1]) : 1;
+	bool ret = rz_core_debug_step_back(core, times);
+	rz_core_dbg_follow_seek_register(core);
+	return ret ? RZ_CMD_STATUS_OK : RZ_CMD_STATUS_ERROR;
+}
+
+// dsf
+RZ_IPI RzCmdStatus rz_cmd_debug_step_frame_handler(RzCore *core, int argc, const char **argv) {
+	rz_core_debug_step_until_frame(core);
+	rz_core_dbg_follow_seek_register(core);
+	return RZ_CMD_STATUS_OK;
+}
+// dsi
+RZ_IPI RzCmdStatus rz_cmd_debug_step_cond_handler(RzCore *core, int argc, const char **argv) {
+	int n = 0;
+	rz_cons_break_push(rz_core_static_debug_stop, core->dbg);
+	do {
+		if (rz_cons_is_breaked()) {
+			break;
+		}
+		rz_debug_step(core->dbg, 1);
+		if (rz_debug_is_dead(core->dbg)) {
+			core->break_loop = true;
+			break;
+		}
+		rz_core_reg_update_flags(core);
+		n++;
+	} while (!rz_num_conditional(core->num, argv[1]));
+	rz_cons_break_pop();
+	RZ_LOG_ERROR("stopped after %d instructions\n", n);
+	rz_core_dbg_follow_seek_register(core);
+	return RZ_CMD_STATUS_OK;
+}
+
+// dsl
+RZ_IPI RzCmdStatus rz_cmd_debug_step_line_handler(RzCore *core, int argc, const char **argv) {
+	const int times = argc > 1 ? (int)rz_num_math(core->num, argv[1]) : 1;
+	rz_reg_arena_swap(core->dbg->reg, true);
+	step_line(core, times);
+	rz_core_dbg_follow_seek_register(core);
+	return RZ_CMD_STATUS_OK;
+}
+
+// dso
+RZ_IPI RzCmdStatus rz_cmd_debug_step_over_handler(RzCore *core, int argc, const char **argv) {
+	const int times = argc > 1 ? (int)rz_num_math(core->num, argv[1]) : 1;
+	rz_core_debug_step_over(core, times);
```

Comment:
this should probably return a bool similarly to other functions. same for the rest. there should be a way to know if things went well or not.